File: scm-csharp/scheme/primitives/PrimitiveMakeVector.cs
namespace scheme;

public class PrimitiveMakeVector : Primitive
{
    public override string Name()
    {
        return "make-vector";
    }

    public override string Info()
    {
        return
            "Syntax: (make-vector k) (make-vector k fill)\n" +
            "Library: (scheme base)\n" +
            "Description: Returns a newly allocated vector of k elements. If fill is given, every element is initialized to fill; otherwise each element is 0.\n" +
            "Example:\n" +
            "  (make-vector 3 0) => #(0 0 0)\n" +
            "  (make-vector 3 'a) => #(a a a)";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 2);
        int n = IntegerMath.ToInt(arguments[0]);
        object obj = 0;
        if (arguments.Length == 2) obj = arguments[1];

        var result = new object[n];
        Array.Fill(result, obj);
        return result;
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveMakeVector.java
package scheme.primitives;

import scheme.*;

import java.util.Arrays;

public class PrimitiveMakeVector extends Primitive {
    @Override
    public String name() {
        return "make-vector";
    }

    @Override
    public String info() {
        return "Syntax: (make-vector k) (make-vector k fill)\n" +
               "Library: (scheme base)\n" +
               "Description: Returns a newly allocated vector of k elements. If fill is given, every element is initialized to fill; otherwise each element is 0.\n" +
               "Example:\n" +
               "  (make-vector 3 0) => #(0 0 0)\n" +
               "  (make-vector 3 'a) => #(a a a)";
    }
    
    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 2);
        int n = IntegerMath.toInt(arguments[0]);
        Object obj = 0;
        if (arguments.length == 2) obj = arguments[1];

        Object[] result = new Object[n];
        Arrays.fill(result, obj);
        return result;
    }
}
